### PR TITLE
Refactor environment meter into separate objects

### PR DESCRIPTION
**Pull request by pattyshack**

### CODE REVIEW

File: fvm/environment/meter.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+) error {
+	// this method is called on every unit of operation, so
+	// checking the context here is the most likely would capture
+	// timeouts or cancellation as soon as they happen, though
+	// we might revisit this when optimizing script execution
+	// by only checking on specific kind of Meter calls.
+	//
+	// in the future this context check should be done inside the cadence
+	select {
+	case <-meter.ctx.Done():
```
Comment (zhangchiqing) [inline fvm/environment/meter.go:82]: Why MeterComputation needs to check meter.ctx.Done(), but not MeterMemory?

**Review by zhangchiqing — COMMENTED** [review]
Looks good. Just have one question

File: fvm/environment/meter.go (hunk 1 above)
Comment (pattyshack) [reply fvm/environment/meter.go:82]: based on my understanding, this is a hack.  we just it need returning error "quickly enough"

**Review by zhangchiqing — APPROVED** [review]

**Review by SaveTheRbtz — APPROVED** [review]

**Comment by pattyshack** [comment]
bors merge

**Pull request merged**